flac.go

```diff
@@ -62,23 +67,42 @@ type Stream struct {
 // Stream.ParseNext to parse the entire next frame including audio samples.
 func New(r io.Reader) (stream *Stream, err error) {
 	// Verify FLAC signature and parse the StreamInfo metadata block.
-	br := bufio.NewReader(r)
-	stream = &Stream{r: br}
-	isLast, err := stream.parseStreamInfo()
+	//br := bufio.NewReader(r)
+	stream = &Stream{r: r}
+	block, err := stream.parseStreamInfo()
 	if err != nil {
 		return nil, err
 	}
 
-	// Skip the remaining metadata blocks.
-	for !isLast {
-		block, err := meta.New(br)
+	rs, seeker := r.(io.ReadSeeker)
+
+	// Parse the remaining metadata blocks.
+	for !block.IsLast {
+		block, err = meta.New(r)
 		if err != nil && err != meta.ErrReservedType {
 			return stream, err
 		}
-		if err = block.Skip(); err != nil {
-			return stream, err
+
+		if seeker {
+			if err = block.Parse(); err != nil {
+				return stream, err
+			}
+			if seeker && block.Header.Type == meta.TypeSeekTable {
+				stream.seekTable = block.Body.(*meta.SeekTable)
+			}
+		} else {
+			if err = block.Skip(); err != nil {
+				return stream, err
+			}
 		}
-		isLast = block.IsLast
+	}
+
+	if seeker {
+		stream.start, _ = rs.Seek(0, io.SeekCurrent)
+	}
+
+	if seeker && stream.seekTable == nil {
+		return stream, stream.makeSeekTable()
```

Comment:
Perhaps makeSeekTable should be called when Seek is called for the first time?

Reply:
> Perhaps makeSeekTable should be called when Seek is called for the first time?

This sounds like a good approach. And then if we introduce an API such as `InitSeekTable`, then the user could ensure that the seek table is already initialized should they wish before the first call to `Seek`.